akka-sample-sharding-java/killrweather/src/main/java/sample/killrweather/WeatherStation.java

```diff
@@ -0,0 +1,241 @@
+package sample.killrweather;
+
+import akka.actor.typed.ActorRef;
+import akka.actor.typed.ActorSystem;
+import akka.actor.typed.Behavior;
+import akka.actor.typed.PostStop;
+import akka.actor.typed.javadsl.AbstractBehavior;
+import akka.actor.typed.javadsl.ActorContext;
+import akka.actor.typed.javadsl.Behaviors;
+import akka.actor.typed.javadsl.Receive;
+import akka.cluster.sharding.typed.javadsl.ClusterSharding;
+import akka.cluster.sharding.typed.javadsl.Entity;
+import akka.cluster.sharding.typed.javadsl.EntityTypeKey;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Generally geo-based data, unless it is a floating station.
+ *
+ * This is just an example: cluster sharding would be overkill for just keeping a small amount of data,
+ * but becomes useful when you have a collection of 'heavy' actors (in terms of processing or state)
```

Comment:
Maybe remove the heavy section, other reasons for using sharding e.g. resiliency, regardless of how heavy they are